Make key shortcut match description.

alexandermorgan opened the pull request:
Changes keyboard shortcut to toggle notation view from CTRL+ALT+CMD+M to CTRL+ALT+M. The new shortcut works upon restarting Atom, but the shortcut is no longer displayed when you click on Packages>MEI Tools. I'm not sure why that is.

alexandermorgan committed:
Make key shortcut match description.

Changes keyboard shortcut to toggle notation view from CTRL+ALT+CMD+M to CTRL+ALT+M.

siennamw commented:
Hi Alex!

"Make key shortcut match description" <- what description are you referring to? Everything I'm seeing says `ctrl-alt-cmd-m`.

alexandermorgan commented:
I should have suspected this was a Windows/Mac thing since it uses CMD. Here's a screenshot that shows that the keymap is set up like you had it, but the description in the pull-down "Packages" menu says "ALT-CTRL-M". They talk a bit more about this on these issues in Atom:
[1](https://discuss.atom.io/t/dumb-question-why-is-control-denoted-cmd-in-the-docs/10737
)
[2](https://discuss.atom.io/t/cmd-key-on-windows/19723
)
I think if we stay away from the Mac CMD button it would make things a little simpler. Also, the shortcut becomes 3 buttons instead of 4, and since it's the only shortcut it doesn't cause any collisions.

![image](https://user-images.githubusercontent.com/2601689/42832786-56e33d54-89c0-11e8-8fed-ea133c2d6e7d.png)

alexandermorgan committed:
Add functionality to generate xml:ids.

For this to work, lines 104 and 107 need to be replaced with the correct calls to get and set the file text.

siennamw commented:
Going forward, it would be great to have separate pull requests for different features.  So, one for changing the shortcut keys and one for adding the XML:ID functionality.  But this is fine for now.

As for the part where you get and set the contents of a file, how about following the pattern I used for conversion from MusicXML to MEI?  This reads in the content of the current file, runs it through the converter, and writes the output to a brand new file.  Perhaps the new file can have something like '-generated-ids' appended to its name.

Also, please update the readme accordingly.

alexandermorgan committed:
WIP

alexandermorgan commented:
I can separate out the PRs so that they contain one change each and also update the readme, no problem. I still have two issues though. I'm not sure what you want to do about the keymap issue, and more importantly I'm still having a hard time understanding how to get and set the text of a file. this.activeEditor.getText() and setText() would be the obvious functions but they don't seem to be working. It's a little tough since there's no documentation on any of the components of this repository. I tried basing it off of your conversion from xml to mei but so far with no success. It doesn't help that I'm not very strong reading Javascript. If you could please just tell me how to call these two functions and where they are it would be really helpful.

siennamw commented:
I suggest that you use this function as a model: https://github.com/siennamw/mei-tools-atom/blob/master/lib/mei-tools-atom.js#L78.  `getText()` is called here https://github.com/siennamw/mei-tools-atom/blob/master/lib/mei-tools-atom.js#L78, and `setText()` is called here https://github.com/siennamw/mei-tools-atom/blob/master/lib/mei-tools-atom.js#L100. You can even reuse the `resolveNewFileName` function.  I hope that clears things up.  Let me know if you have any more questions.

siennamw commented:
Also, it's fine with me to change the shortcut to `ALT-CTRL-M` as you suggested.  I'm a little concerned about collisions with other packages, but we can deal with that later if it comes up.

alexandermorgan commented:
I'm going to make a new PR with its own branch.
